modules/account-baseline/modules/baseline-cloudtrail/kms.tf

```diff
@@ -0,0 +1,87 @@
+data "aws_iam_policy_document" "kms" {
+  statement {
+    sid     = "EnableIAMUserPermissions"
+    effect  = "Allow"
+    actions = ["kms:*"]
+    principals {
+      type        = "AWS"
+      identifiers = ["arn:aws:iam::${data.aws_caller_identity.current.account_id}:root"]
+    }
+    resources = ["*"]
+  }
+  statement {
+    sid     = "AllowCloudTrailEncryption"
+    effect  = "Allow"
+    actions = ["kms:GenerateDataKey*"]
+    principals {
+      type        = "Service"
+      identifiers = ["cloudtrail.amazonaws.com"]
+    }
+    resources = ["*"]
+    condition {
+      test     = "StringLike"
+      variable = "kms:EncryptionContext:aws:cloudtrail:arn"
+      values   = ["arn:aws:cloudtrail:*:${data.aws_caller_identity.current.account_id}:trail/*"]
+    }
+  }
+  statement {
+    sid     = "AllowCloudTrailDescribeKey"
+    effect  = "Allow"
+    actions = ["kms:DescribeKey"]
+    principals {
+      type        = "Service"
+      identifiers = ["cloudtrail.amazonaws.com"]
+    }
+    resources = ["*"]
+  }
+  statement {
+    sid     = "AllowPrincipalDecryption"
+    effect  = "Allow"
+    actions = ["kms:Decrypt", "kms:ReEncryptFrom"]
+    principals {
+      type        = "Service"
+      identifiers = ["cloudtrail.amazonaws.com"]
+    }
+    resources = ["*"]
+    condition {
+      test     = "StringEquals"
+      variable = "kms:CallerAccount"
+      values   = [data.aws_caller_identity.current.account_id]
+    }
+    condition {
+      test     = "StringLike"
+      variable = "kms:EncryptionContext:aws:cloudtrail:arn"
+      values   = ["arn:aws:cloudtrail:*:${data.aws_caller_identity.current.account_id}:trail/*"]
+    }
+  }
+  statement {
+    sid     = "AllowCloudWatchLogsGroup"
+    effect  = "Allow"
+    actions = ["kms:Encrypt*", "kms:Decrypt*", "kms:ReEncrypt*", "kms:GenerateDataKey*", "kms:Describe*"]
+    principals {
+      type        = "Service"
+      identifiers = ["logs.${data.aws_region.current.name}.amazonaws.com"]
+    }
+    resources = ["*"]
+    condition {
+      test     = "ArnEquals"
+      variable = "kms:EncryptionContext:aws:logs:arn"
+      values   = ["arn:aws:logs:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:log-group:${local.cloudwatch_log_group_name}"]
```

Comment:
@chelnak you'll notice that this uses a local reference instead of a resource attribute for the log group name - This was a 🐔 🥚 🐣 ... I was trying to generate a policy with a reference to an object that was a dependent of this one and it was giving me a cycle error, something I've never seen before

Reply:
I've had the same happen!